FAQ updated (#8356)
harness/developer-hub: kb/continuous-integration/continuous-integration-faqs.md

No. You must upgrade the Harness Docker runner manually.

###  Why the CI stage execution is failing with the error "Error response from daemon: could not find an available, non-overlapping IPv4 address pool among the defaults to assign to the network" when running the build on local build infra?

> This could happen when there are many stale docker networks and the runner is unable to create any new network. You could run the command "docker network prune" to delete the unused stale networks on the machine wher the runner is running

Reviewer: ```suggestion
This could happen when there are many stale docker networks and the runner is unable to create any new network. You could run the command "docker network prune" to delete the unused stale networks on the machine wher the runner is running.
```

### Does runner configured in the local build infra creates docker network as part of each execution?

> Yes, the runner will be creating the docker networks as part of each CI stage execution and these networs will be deleted once the execution got completed

Reviewer: ```suggestion
Yes, the runner will be creating the docker networks as part of each CI stage execution and these networks will be deleted once the execution got completed.
```

### When we run the CI stage using local build infrastructure, where does the runner execute each step? Does it run as a process on the host machine, or does it execute within a container?

> The native harness steps will be running within a container by default and the run step can be configured to run on the host machine or run inside a container

Reviewer: ```suggestion
The native harness steps will be running within a container by default and the run step can be configured to run on the host machine or run inside a container.
```

### Is the host machine auto cleaned up after executing the CI stage on local build infrastructure, similar to how it is when running the build on a VM?

> No, host machine doesnt get cleaned up when the build is running on local build infra

Reviewer: ```suggestion
No, host machine doesn't get cleaned up when the build is running on local build infrastructure.
```

- [Verify connectivity for GCP VM build infra](https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/vm-build-infrastructure/define-a-ci-build-infrastructure-in-google-cloud-platform#verify-connectivity)
- [Verify connectivity for Anka macOS VM build infra](https://developer.harness.io/docs/continuous-integration/use-ci/set-up-build-infrastructure/vm-build-infrastructure/define-macos-build-infra-with-anka-registry#verify-connectivity)

### When we run the onprem VM build on windows machine, the windows VM takes more time to start compared to the linux VM. Is there way to speed up the windows VM start up?

> Windows VMs would usually take more time to start compared to linux VM. You could keep a warm VM always running ready to accept the incoming build request by configuring pool with 1 or higher number in the pool.yaml

Reviewer: ```suggestion
Windows VMs would usually take more time to start compared to linux VM. You could keep a warm VM always running ready to accept the incoming build request by configuring pool with 1 or higher number in the `pool.yaml`.
```

### After making changes in the pool.yml, why the runner is not taking the new changes?

> This might be because the runner wasn't restarted after updating the pool.yml. The runner must be restarted for the changes in pool.yml to take effect

Reviewer: ```suggestion
This might be because the runner wasn't restarted after updating the `pool.yml`. The runner must be restarted for the changes in `pool.yml` to take effect.
```

Yes

### Im configuring the hostpath in the build stage to mount a folder from the delegate to the build pod. Why can we not see the folder from delegate inside the step container during execution?

> The host path mount will copy the contents of the folder from the node where the pod is running not from the delegate. It will not pick the files/folder from the delegate and there is currently not a built-in way to copy the contents from the delegate to the build pod

Reviewer: ```suggestion
The host path mount will copy the contents of the folder from the node where the pod is running not from the delegate. It will not pick the files/folder from the delegate and there is currently not a built-in way to copy the contents from the delegate to the build pod.
```
